api/v1alpha1/openapi.yaml

```diff
@@ -1993,20 +1993,29 @@ components:
           description: 'Current state of the device.'
           items:
             $ref: '#/components/schemas/Condition'
-        systemInfo:
+        systemInfo: # deprecated
           $ref: '#/components/schemas/DeviceSystemInfo'
-        containers:
+        containers: # deprecated
           type: array
           description: Statuses of containers in the device.
           items:
             $ref: '#/components/schemas/ContainerStatus'
-        systemdUnits:
+        systemdUnits: # deprecated
           type: array
           description: 'Current state of systemd units on the device.'
           items:
             $ref: '#/components/schemas/DeviceSystemdUnitStatus'
+        workload:
+          $ref: '#/components/schemas/DeviceWorkloadStatus'
+          description: 'Current status of the device workload.'
+        system:
+          $ref: '#/components/schemas/DeviceSystemStatus'
+          description: 'Current status of the device system.'
+        config:
```

Comment:
This is a good question. In general mirroring is not the case because it's not required. For the reasons I stated about a renderedVersion being like a transaction a single failure means none of the status values will be updated for fields provided by spec.

Also because we have one giant object there is no way for partial sync. For example only add a workload unless that is all that is added.

So what value is there to reflect the values defined by the spec and gated by the renderedVersion? I see little. 

If there is a failure at its logical level it will be summarized. Providing a reason for failure and info to help debug. Also during runtime if a workload fails in the steady state we get that information.

Reply:
One area of clarification would be what are the responsibilities with regards to status if the device is unmanaged